### PR TITLE
Floats can be created from numpy floats

### PR DESCRIPTION
**Pull request by cbm755**
This is rebasing the work of @bmgxyz and @Trave11er who both worked on the same issue #13275.

Also, update the AUTHORS file.  @bmgxyz and @Trave11er please confirm that this is how you would like your name/email/etc.

I'm not planning on merging this until all involved are happy.  And I can rebase it if there is further changes at #13400 and #13401.

### CODE REVIEW

**Review by bmgxyz — APPROVED** [review]
Looks good to me. If we end up needing doctests for ```_convert_numpy_types()```, then we can deal with that in the future.

@cbm755 Thanks for taking the time to combine our work. :+1:

**Comment by Trave11er** [comment]
Indeed - thank you, @cbm755 for reviewing our code and putting it all together.

One small thing - in test_numbers.py the function name is `check_prec_and_relerr` but the call is for `check_precision_and_relerr`. Otherwise, looks good.

**Comment by cbm755** [comment]
Nice catch @Trave11er; I've fixed and rebased. 

So then why did the tests pass?  Travis never runs with numpy apparently.

**Comment by cbm755** [comment]
@asmeurer I'm a bit too close to this now, can you review these tests?  I'm +1 for the other changes.

**Comment by ylemkimon** [comment]
> So then why did the tests pass? Travis never runs with numpy apparently.

Travis runs only selected tests with NumPy. Could you add `sympy/core/tests/test_numbers.py` to it?
https://github.com/sympy/sympy/blob/0cfd9a739c9b8095cbd8d42fe154af4634c7841d/bin/test_travis.sh#L80-L83

**Comment by asmeurer** [comment]
Looks good. Can someone double check that the tests pass on 32-bit?

File: sympy/core/tests/test_numbers.py [hunk 1]
```diff
@@ -1749,3 +1749,32 @@ def test_Integer_precision():
...
+        assert x._prec == prec
+        y = Float(ratval, precision=prec)
+        assert abs((x - y)/y) < 2**(-(prec+1))
+
+    check_prec_and_relerr(np.float32(2)/3, S(2)/3)
+    check_prec_and_relerr(np.float64(2)/3, S(2)/3)
+    check_prec_and_relerr(np.float128(2)/3, S(2)/3)
+
+    # float128 more precise than double (how much more is arch dep)
+    x = np.float128(2)/3
```
Comment (ylemkimon) [inline sympy/core/tests/test_numbers.py]: `float128` may not be available exist on all platforms. You should do attribute check or remove it.
Fails on Windows 32-bit: [Appveyor](https://ci.appveyor.com/project/ylemkimon/sympy/build/1.0.16).

**Review by ylemkimon — CHANGES_REQUESTED** [review]

**Comment by ylemkimon** [comment]
@asmeurer I think there should be a Windows testing, e.g., AppVeyor, for at least optional dependencies(native CPython extensions). During testing of above PR, I found sympify test fails on Windows 64-bit(#13454).

**Comment by asmeurer** [comment]
Agreed

File: sympy/core/tests/test_numbers.py (hunk 1 above)
Comment (cbm755) [reply sympy/core/tests/test_numbers.py]: Thanks for this!  [The docs](https://docs.scipy.org/doc/numpy/user/basics.types.html) say:

>  In spite of the names, np.float96 and np.float128 provide only as much precision
>  as np.longdouble, that is, 80 bits on most x86 machines and 64 bits in standard Windows builds.

So yes this particular test should be removed.

However, the docs also say:

>  np.float96 and np.float128 are provided for users who want specific padding.

But you say `float128` may not be available.  And indeed `np.float96` is not available on my GNU/Linux 64-bit!  I guess I'll change the test to `longdouble`.

**Comment by cbm755** [comment]
@ylemkimon thanks for the feedback, I think I've addressed your review.  Ok to merge this?

**Review by ylemkimon — APPROVED** [review]

**Pull request merged**